Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1102,6 +1106,38 @@ namespace Babylon
         LoadTextureFromImage(texture, image, false);
     }
 
+    void NativeEngine::LoadRawTexture2DArray(const Napi::CallbackInfo& info)
+    {
+        const auto texture{info[0].As<Napi::Pointer<TextureData>>().Get()};
+        const auto data = info[1].As<Napi::TypedArray>();
+        const auto width{static_cast<uint16_t>(info[2].As<Napi::Number>().Uint32Value())};
+        const auto height{static_cast<uint16_t>(info[3].As<Napi::Number>().Uint32Value())};
+        const auto depth{static_cast<uint16_t>(info[4].As<Napi::Number>().Int32Value())};
+        const auto format{static_cast<bimg::TextureFormat::Enum>(info[5].As<Napi::Number>().Uint32Value())};
+        const auto generateMips = info[6].As<Napi::Boolean>().Value();
+        const auto invertY = info[7].As<Napi::Boolean>().Value();
+        const auto samplingMode = info[8].As<Napi::Number>().Int32Value();
+        const auto textureType = info[10].As<Napi::Number>().Int32Value();
+
+        texture->Width = width;
+        texture->Height = height;
+        texture->CreationFlags = BGFX_TEXTURE_NONE;
+        texture->Flags = BGFX_TEXTURE_NONE | BGFX_SAMPLER_NONE | BGFX_CAPS_TEXTURE_2D_ARRAY;
+        texture->Handle = bgfx::createTexture2D(width, height, generateMips, depth, Cast(format), texture->Flags);
+        texture->OwnsHandle = true;
+
+        if (!data.IsNull())
+        {
+            if (data.ByteLength() != bimg::imageGetSize(nullptr, width, height, 1, false, false, depth, format))
+                throw std::runtime_error{"The data size does not match width, height, depth and format"};
+
+            uint8_t* dataPtr = static_cast<uint8_t*>(data.ArrayBuffer().Data()) + data.ByteOffset();
+            uint32_t dataSize = static_cast<uint32_t>(data.ByteLength());
+            const bgfx::Memory* dataCopy = bgfx::copy(dataPtr, dataSize); // This is required since BGFX must manage the data the memory.
+            bgfx::updateTexture2D(texture->Handle, 0, 0, 0, 0, width, height, dataCopy);
```

Comment:
> // This is required since BGFX must manage the data the memory.

I'm not sure this is true. We could use bgfx::makeRef instead and hold on to the JS memory until it is used.